shared/constants/chat.js

```diff
@@ -1121,13 +1118,58 @@ const getUserItems = createShallowEqualSelector(
       .toArray()
 )
 
+// Selectors for entities
+function getConversationMessages(state: TypedState, convIDKey: ConversationIDKey): KBOrderedSet<MessageKey> {
+  return state.entities.conversationMessages.get(convIDKey, OrderedSet())
+}
+
+function getDeletedMessageIDs(state: TypedState, convIDKey: ConversationIDKey): Set<MessageID> {
+  return state.entities.deletedIDs.get(convIDKey, Set())
+}
+
+function getMessageUpdates(
+  state: TypedState,
+  messageKey: MessageKey
+): KBOrderedSet<EditingMessage | UpdatingAttachment> {
+  const {conversationIDKey, messageID} = splitMessageIDKey(messageKey)
+  const updateKeys = state.entities.messageUpdates.getIn([conversationIDKey, String(messageID)], OrderedSet())
+  return updateKeys.map(k => state.entities.messages.get(k))
+}
+
+function getMessageFromMessageKey(state: TypedState, messageKey: MessageKey): ?Message {
+  return state.entities.messages.get(messageKey)
+}
+
+function applyMessageUpdates(message: Message, updates: KBOrderedSet<EditingMessage | UpdatingAttachment>) {
```

Comment:
Code that takes a message, and an ordered set of updates and reduces the updates into a new message